app/src/main/java/com/truongdc21/mediatree/intro/SplashActivity.kt

```diff
@@ -0,0 +1,51 @@
+package com.truongdc21.mediatree.intro
+
+import android.content.Context
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.preferencesDataStore
+import androidx.lifecycle.lifecycleScope
+import com.truongdc21.mediatree.MainActivity
+import com.truongdc21.mediatree.R
+import com.truongdc21.mediatree.base.BaseActivity
+import com.truongdc21.mediatree.databinding.ActivitySplashBinding
+import com.truongdc21.mediatree.utils.*
+import kotlinx.coroutines.delay
+
+val Context.dataStore: DataStore<Preferences> by preferencesDataStore(name = ConstantPreferences.dataStore)
+class SplashActivity : BaseActivity<ActivitySplashBinding>(ActivitySplashBinding::inflate) {
+
+    override fun initViews() {
+        checkFirstInstall()
+    }
+
+    override fun initData() {}
+
+    private fun checkFirstInstall() = lifecycleScope.launchWhenCreated {
+        val isCheck : Boolean? = dataStore.readPreferencesFirstInstall()
```

Comment:
nên lưu default value = false thay vì check null

Reply:
Tối thấy làm vậy nó hơi bị rối ấy ông , tại vì mặc đính của thằng này là null , với lại mục đích là check null chứ không hẳn là check true false

Reply:
ủa thế là đặt biến boolean xong check null thôi ấy hả :v

Reply:
giả sử như này .. nếu trong hàm checkFirstInstall đó ,,  tối set isCheck đó default là false , và tôi check trong hàm if nếu fales thì hiện màn hình instalFirst else thì vào main .. dạ xử lần sau tôi quay lại app thì rõ ràng nó lại set default là false và nó lại vào màn hình instalFirst ..

Reply:
thì sau khi. ông vào màn hình installfirst thì phải set lại = true chứ , sao để nó check mãi vậy đc

Reply:
var isCheck = false
        if (!isCheck) {
             // -> vào install first
        }else {
            //  -> vào main và set isCheck = true
        }
--> vì khi gọi cái hàm này đầu tiên mình sẽ set isCheck nó là = false
--> khi chạy vào hàm else --> nó lại set isCheck - true
nhưng mà mở lại app thì nó vẫn sẽ gọi lại hàm thì nó lại set isCheck bằng false và nó tiếp tục vào instalFirst

Reply:
Vơi lại cái dataStore đó có thể null nên là check Null là an toàn nhất